walkthroughs/howto-cross-ekscluster-appmesh-with-cloudmap/README.md

```diff
@@ -0,0 +1,232 @@
+```
+
+*Install AppMesh Components*
+
+Run the following set of commands to install the AppMesh controller and Injector components 
+
+```
+helm repo add eks https://aws.github.io/eks-charts
+kubectl create ns appmesh-system
+kubectl apply -f https://raw.githubusercontent.com/aws/eks-charts/master/stable/appmesh-controller/crds/crds.yaml
+helm upgrade -i appmesh-controller eks/appmesh-controller --namespace appmesh-system
+helm upgrade -i appmesh-inject eks/appmesh-inject --namespace appmesh-system --set mesh.create=true --set mesh.name=global
+
+Opitionally add tracing
+helm upgrade -i appmesh-inject eks/appmesh-inject --namespace appmesh-system --set tracing.enabled=true --set tracing.provider=x-ray
+```
+
+We are now ready to deploy our front and colorapp applications to respective clusters along with the appmesh which will span both clusters.
+
+
+## Setup
+
+1. You can run all commands from this location
+   ```
+   walkthroughs/howto-cross-ekscluster-appmesh-with-cloudmap/
+   ```
+2. **Your** account id:
+    ```
+    export AWS_ACCOUNT_ID=<your_account_id>
+    ```
+3. **Region** e.g. us-west-2
+    ```
+    export AWS_DEFAULT_REGION=us-west-2
+    ```
+4. **ENVOY_IMAGE** environment variable is set to App Mesh Envoy, see https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html
+    ```
+    export ENVOY_IMAGE=...
+    ```
+5. **VPC_ID** environment variable is set to the VPC where Kubernetes pods are launched. VPC will be used to setup private DNS namespace in AWS using create-private-dns-namespace API. To find out VPC of EKS cluster you can use `aws eks describe-cluster`. See [below](#1-how-can-i-use-cloud-map-namespaces-other-than-privatednsnamespace) for reason why Cloud Map PrivateDnsNamespace is required.
+    ```
+    export VPC_ID=...
+    ```
+6. **CLUSTER** environment variables are set
+    ```
+    export CLUSTER1=<cluster name>
+    export CLUSTER2=<cluster name>
+    ```
+7. Deploy
+    ```. 
+    ./deploy.sh
+    ```
+
+## Verify CloudMap and Mesh
+
+As a part of deploy command we have pushed the images to ECR, created a namespace in CloudMap and created the mesh and the DNS entries by virtue of adding the service discovery attributes.
+
+You may verify this, with the following command:
+```
+aws servicediscovery discover-instances --namespace appmesh-demo.pvt.aws.local
+ --service-name colorapp
+```
+This should resolve to the backend service.
+
+You can verify under AppMesh console to verify that the virtual nodes, virtual services, virtual router and routes are indeed created.
+
+## Test the application
+
+The front service in cluster1 has been exposed as a loadbalancer and can be used directly 
+```
+>>kubectl get svc -n appmesh-demo
+NAME    TYPE           CLUSTER-IP      EXTERNAL-IP                                                               PORT(S)        AGE
+front   LoadBalancer   10.100.145.29   af3c595c8fb3b11e987a30ab4de89fc8-1707174071.us-east-1.elb.amazonaws.com   80:31646/TCP   5h47m
```

Comment:
not sure why we need an ALB setup as part of this walkthrough. Can it be removed?

Reply:
I think it's better to leave this for folks who want to test quickly instead of using a curler pod. Does not really matter.

Reply:
you don't need to use curler pod if we can show using `kubectl port-forward`